include/core/serial_solid.h

```diff
@@ -0,0 +1,210 @@
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Authors: Carole-Anne Daunais, Valérie Bibeau, Polytechnique Montreal, 2020
+ * Jeanne Joachim, Polytechnique Montreal, 2020-
+ */
+
+#ifndef lethe_serial_solid_base_h
+#define lethe_serial_solid_base_h
+
+
+// Lethe Includes
+#include <core/parameters.h>
+#include <core/pvd_handler.h>
+#include <core/simulation_control.h>
+#include <core/solid_objects_parameters.h>
+
+// Dealii Includes
+#include <deal.II/base/function.h>
+#include <deal.II/base/quadrature_lib.h>
+
+#include <deal.II/distributed/tria_base.h>
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/mapping_fe.h>
+
+#include <deal.II/lac/trilinos_vector.h>
+
+#include <deal.II/particles/particle_handler.h>
+
+
+using namespace dealii;
+
+/**
+ * A base class that generates a particle handler for the solid
+ *
+ * @tparam dim An integer that denotes the dimension of the space in which
+ * the flow is solved
+ * @tparam spacedim An integer that denotes the dimension of the space occupied
+ * by the embedded solid
+ *
+ * @author Carole-Anne Daunais, Valerie Bibeau, 2020
+ */
+
+template <int dim, int spacedim = dim>
+class SerialSolid
+{
+public:
+  // Member functions
+  SerialSolid(std::shared_ptr<Parameters::RigidSolidObject<spacedim>> &param,
+              unsigned int                                             id);
+
+  /**
+   * @brief Manages solid triangulation and particles setup
+   */
+  void
+  initial_setup();
+
+  /**
+   * @brief Generates a solid triangulation from a dealii or gmsh mesh
+   */
+  void
+  setup_triangulation(const bool restart);
+
+  /**
+   * @brief Set-ups the displacement vector to store the motion of the triangulation
+   */
+  void
+  setup_displacement();
+
+  /**
+   * @brief Loads a solid triangulation from a restart file
+   */
+  void
+  load_triangulation(const std::string filename_tria);
+
+  /**
+   * @return shared_ptr of the solid triangulation
+   */
+  std::shared_ptr<Triangulation<dim, spacedim>>
+  get_solid_triangulation();
+
+  /**
+   * @return the reference to the solid dof handler
+   */
+  DoFHandler<dim, spacedim> &
+  get_solid_dof_handler();
+
+  /**
+   * @return the reference to the displacement dof handler
+   */
+  DoFHandler<dim, spacedim> &
+  get_displacement_dof_handler();
+
+  /**
+   * @return the reference to the displacement vector
+   */
+  Vector<double> &
+  get_displacement_vector();
+
+  /**
+   * @return Function<spacedim> of the solid velocity
+   */
+  Function<spacedim> *
+  get_solid_velocity();
```

Comment:
I don't think this function is needed anymore no?
We could remove it alltogether.

Reply:
I haven't still added the parameters section. I will remove it in the end if it was unnecessary.

Reply:
done.